setup/mysql.sql

```diff
@@ -198,12 +205,18 @@ ALTER TABLE `domains` ENABLE KEYS;
 UNLOCK TABLES;
 
 --
+-- Generate password for siteadmin
+--
+
+SET @SITE_ADMIN_PW = SUBSTRING(TO_BASE64(LOWER(CONV(FLOOR(RAND() * 99999999999999), 10, 36))),1,10);
+
+--
 -- Seed the `users` table with the siteadmin user
 --
 
 LOCK TABLES `users` WRITE;
 ALTER TABLE `users` DISABLE KEYS;
-INSERT INTO `users` VALUES (1,1,'siteadmin','siteadmin','$6$uR.EiB1dj5rrvwMF$Qh5LgdjOZavKXwhi9IF0Yuzu7qxsG.dLTTB8e./55ZRNfBuZVLnfUSOEXa0oWT6174myO.WYkOy83HYWAKPbK/',65535,65535,'','','site',1,0,0,0,0,0,0,0,1,NULL,NULL,0,0,0,'SiteAdmin',0,0,NULL,NULL);
+INSERT INTO `users` VALUES (1,1,'siteadmin','siteadmin',ENCRYPT(@SITE_ADMIN_PW,CONCAT(@PW_PREFIX,MD5(RAND()))),65535,65535,'','','site',1,0,0,0,0,0,0,0,1,NULL,NULL,0,0,0,'SiteAdmin',0,0,NULL,NULL);
```

Comment:
> I think this is quite secure already, and we encourage the user to change the password anyway.

There is a risk that the user will keep this password because it "looks" secure.



> If we went with the MD5 hash, we should probably add a call in Vexim to update the hash upon login if its prefix does not match what's in the settings.

If we add this to vexim, also user passwords would be updated, perhaps not a bad idea? On the other side, a sha512-pw with a longer hash does require only some work on the mysql-creation-scripts, so less testing.



> ENCRYPT() is deprecated. Although this does not have immediate effect, we should not forget this.

For vexim3, I'd like to see some kind of installer-script in php. So it creates the database for you.